Sources/Prelude/Refinements.swift

```diff
@@ -0,0 +1,136 @@
+//
+// This source file is part of Prelude, an open source project by Wayfair
+//
+// Copyright (c) 2018 Wayfair, LLC.
+// Licensed under the 2-Clause BSD License
+//
+// See LICENSE.md for license information
+//
+
+/// protocol that represents a rule that can be used to “refine” values of type `RefinedType`. Implement the `static Refinement.isValid(_:)` function to describe which values are allowed.
+public protocol Refinement {
+    /// the type being “refined”
+    associatedtype RefinedType
+
+    /// implement this function to describe which values of type `RefinedType` should be allowed
+    ///
+    /// - Parameter value: a value of type `RefinedType`
```

Comment:
The rest of the comments here use the terminology “proposed value”… I should change this one too